pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -712,7 +712,9 @@ func (m *kubeGenericRuntimeManager) doPodResizeAction(pod *v1.Pod, podStatus *ku
 			}
 		}
 		if newPodCgLimValue < currPodCgLimValue {
-			err = setPodCgroupConfig(rName, true)
+			if err = setPodCgroupConfig(rName, true); err != nil {
+				return err
+			}
```

Comment:
As we're here, would it make sense to optimise the code a little bit by combining some checks, e.g.
```
if newPodCgLimValue != currPodCgLimValue {
    if err = setPodCgroupConfig(rName, true); err != nil {
        return err
    }
}```
and the same with newPodCgReqValue comparisons.

Reply:
I don't think we can combine the checks. Please see the comments above:
https://github.com/kubernetes/kubernetes/blob/696ad198013c116c1e871632f463edddc5b909b5/pkg/kubelet/kuberuntime/kuberuntime_manager.go#L694-L695

Reply:
ok, thanks.
/lgtm

/assign @mrunalp @derekwaynecarr @dchen1107 
for approval

Reply:
so the change here is that the same error will be returned, only we will not attempt to adjust the request? Is there any indication it wasn't left like this on purpose?

Reply:
As far as I read this comment, it doesn't look intentional:
https://github.com/kubernetes/kubernetes/blob/e30d994129ea895a9b0180db350e1ad1e2495725/pkg/kubelet/kuberuntime/kuberuntime_manager.go#L696-L697

However, it might be better to try updating both limits and requests.

Reply:
note: At this point, containers have been resized. So, the pod cgroup shrinking will not be retried later because it is decided based on containers’ resources.

Reply:
If pod limits succeed in shrinking and pod requests fail to shrink, it may cause a case where “limits < requests”.  So, requests should shrink prior to limits. Then, if requests fail to shrink, limits should not shrink. More precisely, if requests fail to shrink and current requests are larger than new limits, limits should not shrink.

Reply:
It might be better to unify both changes for limits and requests into single `pcm.SetPodCgroupConfig()`.

Reply:
> It might be better to unify both changes for limits and requests into single `pcm.SetPodCgroupConfig()`.

Agreed. Let's save that for a separate change though.